src/lib/idb.ts

```diff
@@ -374,97 +273,85 @@ export default class IDBStorage {
-      }
 
-      return new Promise((resolve, reject) => {
-        const timeout = setTimeout(() => {
-          this.log.error('get request not finished!', entryName, request);
-          reject();
-        }, 3000);
-
-        request.onsuccess = function(event) {
-          const result = request.result;
-          if(result === undefined) {
-            reject('NO_ENTRY_FOUND');
-          } /* else if(typeof result === 'string' &&
-            result.substr(0, 5) === 'data:') {
-            resolve(dataUrlToBlob(result));
-          }  */else {
-            resolve(result);
-          }
+    if(!Array.isArray(entryName)) {
+      entryName = [].concat(entryName);
+    }
 
-          clearTimeout(timeout);
-        }
-  
-        request.onerror = () => {
-          clearTimeout(timeout);
-          reject();
-        };
-      });
-    });
+    return this.getObjectStore<T>('readonly', (objectStore) => {
+      return (entryName as string[]).map((entryName) => objectStore.get(entryName));
+    }, 'get: ' + entryName.join(', '));
   }
 
-  public getAll<T>(): Promise<T[]> {
-    return this.openDatabase().then((db) => {
-      //this.log('getAll pre:', fileName);
-
-      try {
-        const transaction = db.transaction([this.storeName], 'readonly');
-        /* transaction.onabort = (e) => {
-          this.log.error('getAll transaction onabort?', e);
-        }; */
-        const objectStore = transaction.objectStore(this.storeName);
-        var request = objectStore.getAll();
-        
-        //this.log.log('IDB getAll:', fileName, request);
-      } catch(err) {
-        this.log.error('getAll error:', err, request, request.error);
-      }
+  private getObjectStore<T>(mode: IDBTransactionMode, objectStore: (objectStore: IDBObjectStore) => IDBRequest | IDBRequest[], log: string) {
+    const perf = performance.now();
 
-      return new Promise((resolve, reject) => {
-        const timeout = setTimeout(() => {
-          this.log.error('getAll request not finished!', request);
-          reject();
-        }, 3000);
+    this.log(log + ': start');
 
-        request.onsuccess = function(event) {
-          const result = request.result;
-          if(result === undefined) {
-            reject('NO_ENTRY_FOUND');
-          } /* else if(typeof result === 'string' &&
-            result.substr(0, 5) === 'data:') {
-            resolve(dataUrlToBlob(result));
-          }  */else {
-            resolve(result);
-          }
+    return this.openDatabase().then((db) => {
+      return new Promise<T>((resolve, reject) => {
+        const transaction = db.transaction([this.storeName], mode);
 
+        transaction.onerror = (e) => {
           clearTimeout(timeout);
-        }
+          reject(transaction.error);
+        };
   
-        request.onerror = () => {
+        transaction.oncomplete = (e) => {
           clearTimeout(timeout);
-          reject();
+
+          this.log(log + ': end', performance.now() - perf);
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MikeOwino/tweb/pullRequest?prid=7329411)
```suggestion
          this.log(log + ": end", performance.now() - perf);
```